src/housing_unit_inventory/helpers.py

```diff
@@ -0,0 +1,472 @@
+
+    Args:
+        address_pts_fc (str): Address point feature class
+        type_column_name (str, optional): Field that defines the points' types. Defaults to 'PtType'.
+        base_address_value (str, optional): The value indicating a base address. Defaults to 'BASE ADDRESS'.
+
+    Returns:
+        pd.DataFrame.spatial: A spatial dataframe without any base addresses
+    """
+
+    address_pts_no_base_df = (
+        pd.DataFrame.spatial.from_featureclass(address_pts_fc) \
+        .query(f'{type_column_name} != @base_address_value')
+        )
+    return address_pts_no_base_df
+
+
+def load_and_clean_owned_unit_groupings(
+    common_areas_fc, common_area_key_column_name, field_mapping=None, unique_key_column='OBJECTID'
+):
+    """Get PUDs and multi-family parcels from common_areas_fc as a dataframe
+
+    Also set IS_OUG to 'Yes' and copy unique_eky_column to to common_area_key_column_name.
+
+    Args:
+        common_areas_fc (str): Path to the common areas featureclass.
+        common_area_key_column_name: Name of column to be created to hold the unique key for each common area.
+        field_mapping (dict): Used to rename specified columns, {'oldname': 'newname'}
+        unique_key_column (str, optional): Column holding unique identifier for common areas. Defaults to 'OBJECTID'.
+            Will be copied to new common_area_column_name column.
+
+    Raises:
+        ValueError: If values in unique_key_column are not unique. Because this column is used later to aggregate
+            parcel information, non-unique values will result in improper aggregation and invalid results.
+
+    Returns:
+        pd.DataFrame.spatial: Spatially enabled dataframe of owned unit groupings
+    """
+
+    common_areas_df = pd.DataFrame.spatial.from_featureclass(common_areas_fc)
+    if not common_areas_df[unique_key_column].is_unique:
+        raise ValueError(f'Unique key column {unique_key_column} does not contain unique values.')
+
+    #: Rename columns if needed
+    if field_mapping:
+        common_areas_df.rename(columns=field_mapping, inplace=True)
+
+    #: Warn if common areas contain empty geometries, then drop the appropriate rows.
+    empty_shape_row_count = common_areas_df[common_areas_df['SHAPE'].isna()][unique_key_column].count()
+    if empty_shape_row_count:
+        warnings.warn(f'{empty_shape_row_count} common area row[s] had empty geometries')
+        common_areas_df.dropna(subset=['SHAPE'], inplace=True)
+
+    common_areas_df[common_area_key_column_name] = common_areas_df[unique_key_column]
+
+    #: Filter out any common areas that aren't PUDs or multi-family areas (industrial, commercial, etc)
+    common_areas_subset_df = common_areas_df[(common_areas_df['SUBTYPE'] == 'pud') |
+                                             (common_areas_df['TYPE'] == 'multi_family')].copy()
+    common_areas_subset_df['IS_OUG'] = 'Yes'
```

Comment:
**nitpick**: if this is a boolean answer of yes or no, a boolean would make more sense.

Reply:
This value gets written to the final feature class as a note to the end user whether the parcel is an OUG or just a normal parcel. The field is currently a string field (based on WFRC's first data submission), so I'd have to convert this to a string at some point before writing it out and `Yes/No` felt less programmy than `True/False`.